src/connection/host_lookup.c

```diff
@@ -0,0 +1,541 @@
+#if defined (HAVE_GETHOSTBYNAME_R_GLIBC)
+int
+gethostbyname_r_uhost (const char *name,
+		       struct hostent *ret, char *buf, size_t buflen, struct hostent **result, int *h_errnop)
+#elif defined (HAVE_GETHOSTBYNAME_R_SOLARIS)
+struct hostent *
+gethostbyname_r_uhost (const char *name, struct hostent *ret, char *buf, size_t buflen, int *h_errnop)
+#elif defined (HAVE_GETHOSTBYNAME_R_HOSTENT_DATA)
+int
+gethostbyname_r_uhost (const char *name, struct hostent *ret, struct hostent_data *ht_data)
+#else
+#error "HAVE_GETHOSTBYNAME_R"
+#endif /* HAVE_GETHOSTBYNAME_R_GLIBC */
+#else /* WINDOWS */
+int
+gethostbyname_r_uhost (const char *name,
+		       struct hostent *ret, char *buf, size_t buflen, struct hostent **result, int *h_errnop)
+#endif				/* HAVE_GETHOSTBYNAME_R */
+{
+#if defined (WINDOWS)
+  return 0;
+#endif
+
+  struct hostent *hp_buf = NULL;
+
+  if (prm_get_bool_value (PRM_ID_USE_USER_HOSTS) == USE_GLIBC_HOSTS)
+    {
+#ifdef HAVE_GETHOSTBYNAME_R
+#if defined (HAVE_GETHOSTBYNAME_R_GLIBC)
+      return gethostbyname_r (name, ret, buf, buflen, result, h_errnop);
+#elif defined (HAVE_GETHOSTBYNAME_R_SOLARIS)
+      return gethostbyname_r (name, ret, buf, buflen, h_errnop);
+#elif defined (HAVE_GETHOSTBYNAME_R_HOSTENT_DATA)
+      return gethostbyname_r (name, ret, &ht_data);
+#else
+#error "HAVE_GETHOSTBYNAME_R"
+#endif
+#endif
+    }
+  else
+    {
+      hp_buf = host_lookup_internal (name, NULL, HOSTNAME_TO_IPADDR);
+    }
+
+#ifdef HAVE_GETHOSTBYNAME_R
+#if defined (HAVE_GETHOSTBYNAME_R_GLIBC)
+  if (hp_buf == NULL)
+    {
+//err_print fprintf (stdout, msgcat_message (MSGCAT_CATALOG_CUBRID, MSGCAT_SET_ERROR, -ER_FAILED));
+      return EFAULT;
+    }
+
+  if (((*result) = (struct hostent *) malloc (sizeof (struct hostent))) == NULL)
+    {
+      return NULL;
+    }
+  memcpy ((void *) ret, (void *) hp_buf, sizeof (struct hostent));
+  memcpy ((void *) *result, (void *) hp_buf, sizeof (struct hostent));
+
+  return 0;
+#elif defined (HAVE_GETHOSTBYNAME_R_SOLARIS)
+  if (hp_buf == NULL)
+    {
+//err_print fprintf (stdout, msgcat_message (MSGCAT_CATALOG_CUBRID, MSGCAT_SET_ERROR, -ER_FAILED));
+      return NULL;
+    }
+
+  memcpy ((void *) ret, (void *) hp_buf, sizeof (struct hostent));
+
+  return ret;
+#elif defined (HAVE_GETHOSTBYNAME_R_HOSTENT_DATA)
+  if (hp_buf == NULL)
+    {
+//err_print fprintf (stdout, msgcat_message (MSGCAT_CATALOG_CUBRID, MSGCAT_SET_ERROR, -ER_FAILED));
+      return -1;
+    }
+
+  memcpy ((void *) ret, (void *) hp_buf, sizeof (struct hostent));
+
+  return 0;
+#else
+#error "HAVE_GETHOSTBYNAME_R"
+#endif /* HAVE_GETHOSTBYNAME_R_GLIBC */
+#endif /* HAVE_GETHOSTBYNAME_R */
+}
+
+int
+getnameinfo_uhost (struct sockaddr *addr, socklen_t addrlen, char *host, size_t hostlen, char *serv, size_t servlen,
+		   int flags)
+{
+  struct hostent *hp = NULL;
+
+  if (prm_get_bool_value (PRM_ID_USE_USER_HOSTS) == USE_GLIBC_HOSTS)
+    {
+      return getnameinfo (addr, addrlen, host, hostlen, NULL, 0, NI_NOFQDN);
```

Comment:
What is the reason that you change the arguments?

`return (ddr, addrlen, host, hostlen, serv, servlen, flags);`

Reply:
This is my mistake. Last one of commits, I fixed the similar mistake but I didn't find this one.

Thank you. I will apply it!